setup.py

```diff
@@ -622,6 +634,11 @@ def build_mx_extension(build_ext, options):
     build_ext.build_extension(mxnet_mpi_lib)
 
 
+def is_mx_cuda():
+    from mxnet import context
+    return context.num_gpus() > 0
```

Comment:
That will not work if we're building Horovod in Docker on the machine that does not have GPUs.  For PyTorch we had to implement dummy compilation to test if CUDA is enabled in the binary.  If MXNet does not offer diagnostic flag to tell if it was built with CUDA support, maybe we can also do a test compilation?

Reply:
@alsrgv Could you elaborate on how you are going to test on the machine without GPUs? Do you mean you will build a docker image with CUDA and run it on a machine without GPU?

Reply:
The use case is to build a docker image with CUDA on the machine without GPU, but run on the machine with GPU.